pkg/server/api/agent/v1/service.go

```diff
@@ -47,7 +55,52 @@ func (s *Service) DeleteAgent(ctx context.Context, req *agent.DeleteAgentRequest
 }
 
 func (s *Service) BanAgent(ctx context.Context, req *agent.BanAgentRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "method not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if req.Id == nil {
+		log.Error("Invalid request: ID is nil")
```

Comment:
s/ID is nil/missing ID/g ?

Reply:
Now that we are switching to `api.IDFromProto` this block is non longer required. I'll remove it.